lib/CTownHandler.h

```diff
@@ -217,6 +217,8 @@ class DLL_LINKAGE CFaction : public Faction
 	std::string getNameTextID() const override;
 
 	bool hasTown() const override;
+	TerrainId getNativeTerrain() const override;
+	EAlignment getAlignment() const override;

```

Comment:
If class got getter method we can make corresponding members private.

If CTownHandler needs access to these members during loading stage then make CTownHandler friend of CFaction

Reply:
moved to #1882